test/libsolidity/gasTests/abiv2_optimised.sol

```diff
@@ -17,9 +17,9 @@ contract C {
 // optimize-yul: true
 // ----
 // creation:
-//   codeDepositCost: 610600
-//   executionCost: 645
-//   totalCost: 611245
+//   codeDepositCost: 624200
```

Comment:
Should we try to reverse this in the end of the suite or make the EVM transform smarter about this? The gas cost increase seems quite harsh....

Reply:
Maybe it's not only that the EVM transform generates better control flow with inline conditions in the loop, but also or maybe even rather stuff similar to https://github.com/ethereum/solidity/pull/7401/files#r323292467

Reply:
Ah ok - my first point is already tracked in https://github.com/ethereum/solidity/issues/7406 in any case.

Reply:
Ok, since the subsequent PRs will reverse and even improve this, it's fine!